src/lib/sedlexing.ml

```diff
@@ -80,37 +78,45 @@ let set_position lexbuf position =
 
 let set_filename lexbuf fname = lexbuf.filename <- fname
 
-let fill_buf_from_gen f gen buf pos len =
-  let rec aux i =
-    if i >= len then len
-    else (
-      match gen () with
-        | Some c ->
-            buf.(pos + i) <- f c;
-            aux (i + 1)
-        | None -> i)
-  in
-  aux 0
+let refill_buf_from_gen gen =
+  let malformed = ref false in
+  fun buf pos len ->
+    if !malformed then raise MalFormed;
```

Comment:
Is this intended to let the lexer return something before raising `MalFormed`? In this case, is the value gonna be meaningful? Can the error be hidden in some cases e.g. if `refill_buf_from_gen` is not called again?

Reply:
The point of this change, extracted in #140, is to delay the malformed exception until we actually try to lex the malformed part of the input. 

> Can the error be hidden in some cases e.g. if refill_buf_from_gen is not called again?

If `refill` is not called again, it means that the lexer doesn't want to read the malformed part of the input, in which case it's expected to not raise Malformed.